test/units/modules/test_iptables.py

```diff
@@ -1,1174 +1,1402 @@
-            'INPUT',
-        ])
+    """Return iptables binary path."""
+    return IPTABLES_CMD
+
+
+@pytest.fixture
+def mock_basic_commands(mocker):
+    """Mock basic commands like get_bin_path and get_iptables_version."""
+    mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.get_bin_path",
+        return_value=get_bin_path(),
+    )
+    mocker.patch("ansible.modules.iptables.get_iptables_version", return_value=IPTABLES_VERSION)
+
+
+def test_without_required_parameters(mocker):
+    """Test module without any parameters."""
+    mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.fail_json",
+        side_effect=fail_json,
+    )
+    set_module_args({})
+    with pytest.raises(AnsibleFailJson) as exc:
+        iptables.main()
+
+    assert exc.value.args[0]["failed"]
+    assert "Failed to find required executable" in exc.value.args[0]["msg"]
+
+
+def test_flush_table_without_chain(mocker, mock_basic_commands):
+    """Test flush without chain, flush the table."""
+    set_module_args(
+        {
+            "flush": True,
+        }
+    )
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command", return_value=(0, "", "")
+    )
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args[0][0]
+    assert first_cmd_args_list[0], IPTABLES_CMD
+    assert first_cmd_args_list[1], "-t"
+    assert first_cmd_args_list[2], "filter"
+    assert first_cmd_args_list[3], "-F"
+
+
+def test_flush_table_check_true(mocker, mock_basic_commands):
+    """Test flush without parameters and check == true."""
+    set_module_args(
+        {
+            "flush": True,
+            "_ansible_check_mode": True,
+        }
+    )
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command", return_value=(0, "", "")
+    )
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 0
+
+
+def test_policy_table(mocker, mock_basic_commands):
+    """Test change policy of a chain."""
+    set_module_args(
+        {
+            "policy": "ACCEPT",
+            "chain": "INPUT",
+        }
+    )
+    commands_results = [(0, "Chain INPUT (policy DROP)\n", ""), (0, "", "")]
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 2
+    first_cmd_args_list = run_command.call_args_list[0]
+    second_cmd_args_list = run_command.call_args_list[1]
+    assert first_cmd_args_list[0][0] == CONST_INPUT_FILTER
+    assert second_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "filter",
+        "-P",
+        "INPUT",
+        "ACCEPT",
+    ]
+
+
+def test_policy_table_no_change(mocker, mock_basic_commands):
```

Comment:
It looks like `test_policy_table_no_change` and `test_policy_table_changed_false` are basically the same test. So they could be collapsed into one function with parametrize.